crates/provider/src/native.rs

```diff
@@ -0,0 +1,601 @@
+                &dst_file_name,
+            )
+            .expect("Failed to copy file");
+        Ok(())
+    }
+
+    async fn wait_node_ready(&mut self, node_name: String) -> Result<(), Box<dyn Error>> {
+        // check if the process is alive after 1 seconds
+        sleep(Duration::from_millis(1000)).await;
+
+        let process_node_name = self.process_map.get_mut(&node_name).unwrap();
+
+        let pid = process_node_name.pid;
+        let log_dir = process_node_name.log_dir.clone();
+
+        let result = self
+            .run_command(
+                vec![format!("ps {}", pid)],
+                NativeRunCommandOptions { allow_fail: true },
+            )
+            .await
+            .expect("Failed to run `ps` command");
+
+        if result.exit_code.code().unwrap() > 0 {
+            let lines: String = self.get_node_logs(node_name).await.unwrap();
+            // Javier - TODO: check how we will log with tables
+            // TODO: Log with a log table
+            // const logTable = new CreateLogTable({
+            //   colWidths: [20, 100],
+            // });
+            // logTable.pushToPrint([
+            //   [decorators.cyan("Pod"), decorators.green(nodeName)],
+            //   [
+            //     decorators.cyan("Status"),
+            //     decorators.reverse(decorators.red("Error")),
+            //   ],
+            //   [
+            //     decorators.cyan("Message"),
+            //     decorators.white(`Process: ${pid}, for node: ${nodeName} dies.`),
+            //   ],
+            //   [decorators.cyan("Output"), decorators.white(lines)],
+            // ]);
+
+            return Err(Box::new(std::io::Error::new(
+                ErrorKind::Other,
+                "An error occured",
+            )));
+        }
+
+        // check log lines grow between 2/6/12 secs
+        let lines_1: RunCommandResponse = self
+            .run_command(
+                vec![format!("wc -l  {}", log_dir)],
+                NativeRunCommandOptions::default(),
+            )
+            .await
+            .expect("Failed to run `wc -l` command");
+        sleep(Duration::from_millis(2000)).await;
+
+        let lines_2: RunCommandResponse = self
+            .run_command(
+                vec![format!("wc -l  {}", log_dir)],
+                NativeRunCommandOptions::default(),
+            )
+            .await
+            .expect("Failed to run `wc -l` command");
+
+        // Javier-TODO: This looks weird and wrong
+        let lines_1_output: u32 = String::from_utf8(lines_1.std_out)
+            .unwrap()
+            .parse::<u32>()
+            .expect("Error while converting 1st time, lines, to u32");
+        let lines_2_output: u32 = String::from_utf8(lines_2.std_out)
+            .unwrap()
+            .parse::<u32>()
+            .expect("Error while converting 2nd time, lines, to u32");
+
+        if lines_2_output > lines_1_output {
+            return Ok(());
+        };
```

Comment:
This is just counting that the logs lines are increasing.

Reply:
Yes... this is what it supposed to do - as taken from the node CLI. There is a comment mentioning:
```rs
 // check log lines grow between 2/6/12 secs
 ```
 but as I put also in comment.. `this looks weird`. Maybe we can discuss it